frontend/src/pages/Overview/ProjectDialog.js

```diff
@@ -50,9 +49,15 @@ const ProjectDialog = props => {
   const { projects, projectToAdd, editDialogShown, creationDialogShown } = props;
   const { displayName } = projectToAdd;
   const changes = compareObjects(projects, projectToAdd);
-  if (changes.deletedProjectedBudgets && changes.deletedProjectedBudgets.length === 0) {
-    delete changes.deletedProjectedBudgets;
-  }
+  const hasChanges =
+    !_isEmpty(changes.displayName) ||
+    !_isEmpty(changes.description) ||
+    !_isEmpty(changes.thumbnail) ||
+    !_isEmpty(changes.projectedBudgets) ||
+    !_isEmpty(changes.additionalData) ||
+    !_isEmpty(changes.tags) ||
+    !_isEmpty(changes.deletedProjectedBudgets);
```

Comment:
no.
`deletedProjectedBudgets` exists in `changes ` as `[]` when no budget has been deleted
So, `_isEmpty(changes)` always returns `false` because there is always this empty array inside

Reply:
This Code was a workaround for this:
`if (changes.deletedProjectedBudgets && changes.deletedProjectedBudgets.length === 0) {
    delete changes.deletedProjectedBudgets;
  }`